x-pack/plugins/maps/public/connected_components/mb_map/draw_control/draw_feature_control/index.ts

```diff
@@ -15,16 +15,18 @@ import {
   ReduxStateProps,
   OwnProps,
 } from './draw_feature_control';
-import { addNewFeatureToIndex, updateEditShape } from '../../../../actions';
+import { addNewFeatureToIndex, deleteFeatureFromIndex, updateEditShape } from '../../../../actions';
 import { MapStoreState } from '../../../../reducers/store';
-import { getEditState } from '../../../../selectors/map_selectors';
+import { getEditState, getLayerById } from '../../../../selectors/map_selectors';
 import { getDrawMode } from '../../../../selectors/ui_selectors';
 
 function mapStateToProps(state: MapStoreState): ReduxStateProps {
   const editState = getEditState(state);
+  const editLayer = getLayerById(editState ? editState.layerId : null, state);
```

Comment:
How about moving check for editState outside of getLayerById so its `const editLayer = editState ? getLayerById(editState.layerId, state) : undefined;`